sql/catalyst/src/main/java/org/apache/spark/sql/util/CaseInsensitiveStringMap.java

```diff
@@ -78,24 +83,22 @@ public String get(Object key) {
 
   @Override
   public String put(String key, String value) {
-    return delegate.put(toLowerCase(key), value);
+    throw new UnsupportedOperationException();
```

Comment:
let's add the error message to say the map is read only.

Reply:
How about showing warning message here?

Reply:
Why do we need a warning message right before throwing an exception?

Reply:
E.g.
```
spark.read.option("foo", "bar").option("fOO", "bar")....
```
This is valid.